### DISCUSSION TITLE
小狼毫助手 weasel-tool 正式发布

### DISCUSSION THREAD

**Discussion by yanhuacuo** [rime/home#1324] [General]
【weasel-tool】正式发布，C++版重构自原 AHK 版；
更稳定，更好用。

https://github.com/yanhuacuo/98wubi-tables/blob/master/%E8%BE%93%E5%85%A5%E6%B3%95%E8%BD%AF%E4%BB%B6/weasel-tool.7z

或

http://98wb.ysepan.com/

## 特色功能

- 在线加词

直接添加词条到扩展词库

- 扩展词库管理器

可视化管理扩展词库

- 快捷路径

「用户目录」、「程序目录」，「日志目录」，更方便折腾。

- 数据备份

「桌面」、「文档」一键达，并以时间戳为标志建立文件夹。

- 进程守护

可以时时护守「小狼毫算法服务」，防止各种意外。

- 开机自启

可以有效避免小狼毫算法服务开机启动慢的问题。

- 数据协同

「从存档加载」可以自动读取「中州韵助手」所监听的方案，并自动用小狼毫助手接管相应配置。

- 更新算法服务

内置最新 rime.dll ，一键发送到桌面，可以自助替换。

- 示范配置

「查看示范」按钮，可以生成一份示范配置到桌面，供其他方案人士比照使用。

- 配置保存

小狼毫助手可以存起自己的配置文件，方便开机自启时做相应动作。

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/0.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/2.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/3.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/4.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/5.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/6.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/7.png)

- ![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/8.png)

**Comment by fxliang** [comment]
![image](https://github.com/rime/home/assets/4023160/1fd249f1-dc36-41e4-b9c2-d9295d4e3173)
有些描述上容易产生误解，竖排文字vertical_text最好和vertical分开。vertical_text有点像vertical 和horizontal的杂交体的状态了，微调一下界面方便用户理解可能会更好一点

还有竖排文字自动换行布置那个选项可以加上
![image](https://github.com/rime/home/assets/4023160/ebc8047a-ebfd-4999-b448-1259ee800c67)

还有就是有些参数之间有关联关系考虑进去

另外方案图标那个除了中文图标，英文图标也是可以设置的，可以考虑加上
![image](https://github.com/rime/home/assets/4023160/62963634-9558-47ae-be16-0b3e5e4c576a)

**Reply by yanhuacuo** [reply]
https://wwqj.lanzoum.com/iXTV00zen1yd

还没有写完事务逻辑，只做好了「进程守护」 ，大佬过过目。

**Reply by yanhuacuo** [reply]
鼠标划过相应的「开关」，有「悬浮提示」

**Reply by yanhuacuo** [reply]
给「英文」添加图标的参数是什么，只看到 schema/ico 这个特定方案图标。

**Reply by fxliang** [reply]
schema/ascii_icon

**Comment by Techince** [comment]
支持64位小狼毫吗？自用的是64位小狼毫版本。

**Reply by yanhuacuo** [reply]
支持

**Reply by Techince** [reply]
那太棒了

**Comment by yanhuacuo** [comment]
## 近日更新

1. 支持扩展词库
2. 支持托盘菜单
3. 全新的互斥变量做单实例运行
4. 支持未安装小狼毫检测
5. 改善在线加词窗口
6. 进程守护自动复位
7. 保存文件时的默认后缀优化
8. 即时生成配置，减少初次使用的障碍
9. 添加字体自选、图标自选入口
10. 添加缺失 weasel.yaml 下的纠错方法
11. 修改范例文件权限为读写
12. 添加清理示范配置的机制
13. 开机启动时，自动隐藏到托盘
14. 新增全半角图标自选
15. 添加配置文件冲突处置模块
16. 自适应 win11 深色或浅色模式
17. 支持接管64位版本小狼毫（0703）
18. 改善缺项提示窗（0703）
19. 改进导入方案失败后的处置方式（0704）
20. 允许扩展词库未加载成功下使用（0704）
21. 添加「清理日志」功能（0713）
22. 添加「从存档刷新数据流」功能（0713）

**Reply by Techince** [reply]
检测不到64位小狼毫。

**Comment by yanhuacuo** [comment]
> 检测不到64位小狼毫。

已添加支持

**Comment by yanhuacuo** [comment]
![PIC](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/7.png)

**Comment by yanhuacuo** [comment]
改进缺项提示

![pic](https://raw.githubusercontent.com/yanhuacuo/Demo_PIC/main/weasel-tool-shot/8.png)

**Reply by fxliang** [reply]
其实不提示问题也不大,有fallback的
```cpp
_RimeGetIntWithFallback(config, "style/label_font_point", &style.label_font_point, "style/font_point");
_RimeGetIntWithFallback(config, "style/comment_font_point", &style.comment_font_point, "style/font_point");

```

**Comment by yanhuacuo** [comment]
近期更新汇总：

1. 添加新选项卡，补齐小狼毫最新参数支持；
2. 新增快速定义候选项序值；
3. 新增深色模式锁定，可以随时锁定任意主题为深色模式；
4. 新增一键定妆，直接生成多种经典样式表；
5. 新增注解界符控制，可以自定义拆分注解界符；

![1](https://github.com/rime/home/assets/3107994/01d58826-b306-4252-9fcd-c7c4c00309c8)

![2](https://github.com/rime/home/assets/3107994/14638dd9-fbaa-4c9a-9043-c53881222657)

![3](https://github.com/rime/home/assets/3107994/e73b02b3-deab-4482-bdd2-5f48de1280e0)

![4](https://github.com/rime/home/assets/3107994/f9f031ae-a29b-400b-ba00-3d5c3126ec00)

![5](https://github.com/rime/home/assets/3107994/9670641e-6dfd-4bbf-8655-5b292940aa53)

![c](https://github.com/rime/home/assets/3107994/0d195c94-20d9-483d-83f7-23417439b296)

![5BA480C3D94471C68EA03BFAF896DDD3](https://github.com/rime/home/assets/3107994/2efe38bc-324d-46a7-ad40-4f082a41f3b9)

**Comment by sci-42ver** [comment]
请问还维护么? 第一条comment的github链接已经失效了

**Reply by yanhuacuo** [reply]
从这里下：http://98wb.ysepan.com/

**Comment by yanhuacuo** [comment]
视频介绍：

https://www.bilibili.com/video/BV1pM4m1R7Xi/?spm_id_from=333.999.0.0

![236BEDA08E2C6AFBE37A4CE0A2378E94](https://github.com/rime/home/assets/3107994/df3f18c6-517d-4a04-aef9-191c07f87744)

![224971617259CCBD61029F87D5B2E206](https://github.com/rime/home/assets/3107994/5fdcf722-1ffb-4298-ad39-fc86e1869b8d)
